util/include/throughput.hpp

```diff
@@ -19,10 +19,59 @@
 #include "assertUtils.hpp"
 
 namespace concord::util {
+
+/** A Duration Tracker allows to track the sum of multiple none-continues time intervals (durations).
+ * This is done by calling start() / pause() multiple times.
+ * If the last call to the tracker is start(): call totalDuration() to get the sum of intervals, including current
+ * still running interval.
+ * If the last call to a tracker is pause(): you may get the sum of intervals from pause() returned
+ * value or call totalDuration() explicitly.
+ * To reset (re-use) the tracker, call reset() and then start().
+ */
+template <typename T>
+class DurationTracker {
+ public:
+  void start() {
+    ConcordAssert(!running_);
+    start_time_ = std::chrono::steady_clock::now();
+    running_ = true;
+  }
+  uint64_t pause() {
```

Comment:
I would expect to see symmetrical methods `start` / `stop`.

Reply:
pause() gives you the option to stop forever or continue again in the future.  I guess for some people the name stop() might make equal confusion.
I wanted to use pause() because it is more clear that we "stop for a moment" but this isn't the final stage of the object, we can choose to continue in the future.

Reply:
Gotcha, please ignore my comment.